apps/web/app/user/[discordID]/page.tsx

```diff
@@ -0,0 +1,205 @@
+import { CalendarPlusIcon } from '@/components/icons/calender-plus'
+import { HeartIcon } from '@/components/icons/heart'
+import { MedalIcon } from '@/components/icons/medal'
+import { Post } from '@/components/post'
+import { getCanonicalUserUrl } from '@/utils/urls'
+import { Metadata } from 'next'
+import { db, sql } from '@nextjs-forum/db/node'
+import { notFound } from 'next/navigation'
+const getLeaderboardPosition = async (discordID: string) => {
+  const result = await db
+    .with('rankedUsers', (db) =>
```

Comment:
This query works but it is reading every row from the table. I tested the following query and the execution plan seems to be better:
```sql
WITH userAnswers AS
  (SELECT "answersCount"
   FROM users
   WHERE "snowflakeId" = $1
)
SELECT 1 +
  (SELECT COUNT(*)
   FROM users
   WHERE "answersCount" >
       (SELECT "answersCount"
        FROM userAnswers)) AS rank;
```